### PR TITLE
feat: emit error for non-resolved relative path

### LINKED ISSUE #1174: Rolldown should throw errors for implicit external modules
Is a module not found in rolldown an external module? 

How to define an external module

from case 
path: /crates/rolldown/tests/esbuild/packagejson/test_package_json_exports_alternatives
test_package_json_exports_alternatives

expect esbuild output in terminal
```sh

✘ [ERROR] Could not resolve "pkg/apples/red.js"

    src/test_package_json_exports_alternatives/src/entry.js:1:21:
      1 │ import redApple from 'pkg/apples/red.js'
        ╵                      ~~~~~~~~~~~~~~~~~~~

  The module "./good-apples/red.js" was not found on the file system:

    src/test_package_json_exports_alternatives/node_modules/pkg/package.json:3:18:
      3 │     "./apples/": ["./good-apples/", "./bad-apples/"],
        ╵                   ~~~~~~~~~~~~~~~~

  You can mark the path "pkg/apples/red.js" as external to exclude it from the bundle, which will
  remove this error and leave the unresolved path in the bundle.

✘ [ERROR] Could not resolve "pkg/books/red"

    src/test_package_json_exports_alternatives/src/entry.js:3:20:
      3 │ import redBook from 'pkg/books/red'
        ╵                     ~~~~~~~~~~~~~~~

  The module "./good-books/red-book.js" was not found on the file system:

    src/test_package_json_exports_alternatives/node_modules/pkg/package.json:4:18:
      4 │     "./books/*": ["./good-books/*-book.js", "./bad-books/*-book.js"]
        ╵                   ~~~~~~~~~~~~~~~~~~~~~~~~

  You can mark the path "pkg/books/red" as external to exclude it from the bundle, which will remove
  this error and leave the unresolved path in the bundle.

```

but now rolldown will bundle process result is successful 

dist/entry.js

```js

// These modules are treated as external. not throw any error or panic 
import { default as redApple } from "pkg/apples/red.js";
import { default as redBook } from "pkg/books/red";

// node_modules/pkg/good-apples/green.js
var green_default = '\u{1f34f}';

// node_modules/pkg/good-books/green-book.js
var green_book_default = '\u{1f4d7}';

// src/entry.js
console.log({
	redApple,
	greenApple:green_default,
	redBook,
	greenBook:green_book_default
});

```

Cause analysis

crates/rolldown_plugin/src/utils/resolve_id_with_plugins.rs
```rust
// line 72
let resolved = resolver.resolve(importer.map(Path::new), request, import_kind)?;
  Ok(resolved.map(|resolved| ResolvedRequestInfo {
    path: resolved.path,
    module_type: resolved.module_type,
    is_external: false,
    package_json: resolved.package_json,
  }))

// Instead of dealing with oxc-resolver errors, we opt for direct passthroughs, 
// whereas oxc returns a not-found error if it can't resolve.
```

crates/rolldown/src/module_loader/normal_module_task.rs
```rust

// line 288

/* 
Here you can clearly see that all the not-found errors from the previous passthrough are treated as external modules. 
 */
  
match resolved_id {
        Ok(info) => {
          ret.push(info);
        }
        Err(e) => match &e {
          ResolveError::NotFound(..) => {
            warnings.push(
              BuildError::unresolved_import_treated_as_external(
                specifier.to_string(),
                self.resolved_path.path.to_string(),
                Some(e),
              )
              .with_severity_warning(),
            );
            ret.push(ResolvedRequestInfo {
              path: specifier.to_string().into(),
              module_type: ModuleType::Unknown,
              is_external: true,
              package_json: None,
            });
          }
          _ => {
            build_errors.push((&dependencies[idx], e));
          }
        },
      }
    }
```

### PR DESCRIPTION
**Pull request by hyf0**
### Description

Closes https://github.com/rolldown/rolldown/issues/1174.

Clean up old issues. The behavior of this PR is already confirmed by the team. I just lack of time to do it.

### CODE REVIEW

**Comment by hyf0** [comment]
* **#3074** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/3074?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/3074?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: crates/rolldown/tests/esbuild/default/external_module_exclusion_relative_path/_config.json [hunk 1]
```diff
@@ -7,5 +7,5 @@
       }
     ]
   },
-  "expectExecuted": false
+  "expectError": true // TODOO(hyf0): Only for pass tests in #3074. Should investigate if it should raise an error.
```
Comment (hyf0) [inline crates/rolldown/tests/esbuild/default/external_module_exclusion_relative_path/_config.json:10]: Some tests are just skipped and needs to investigate why.

File: crates/rolldown/tests/esbuild/importstar/import_star_no_bundle_capture/_config.json [hunk 2]
```diff
@@ -6,6 +6,7 @@
         "name": "entry",
         "import": "entry.js"
       }
-    ]
+    ],
+    "external": ["./foo"]
```
Comment (hyf0) [inline crates/rolldown/tests/esbuild/importstar/import_star_no_bundle_capture/_config.json:10]: Some tests are ok by just adding correct external option.

**Review by IWANABETHATGUY — APPROVED** [review]
Let's open an issue to track all regressed test.

**Pull request merged**